src/player.cpp

```diff
@@ -3208,9 +3209,20 @@ bool player::unload( item &it )
     }
 
     add_msg( _( "You unload your %s." ), target->tname() );
+
+    if( it.has_flag( "MAG_DESTROY" ) && it.ammo_remaining() == 0 ) {
+        loc.remove_item();
+    }
+
     return true;
 }
 
+bool player::unload( item &it )
```

Comment:
Refactoring callers to use `item_location` would be better than a wrapper function.

Reply:
Sure, I'll do that. Some of the functions calling player::unload take an `item` instead of an `item_location` themselves though, so those will have to construct an `item_location` themselves

Reply:
Yeah there's a lot of older code still using `item *` and `int pos` everywhere. We're working on migrating it all to use `item_location`